### ISSUE TITLE
Vscode extensions are put in wrong paths in master with nixpkgs-unstable

### ISSUE THREAD

**Issue by oxalica** [rycee/home-manager#1302]
### Issue description

I'm on home-manager master(a21c97d0113ebc43983ff688b2d9abddd07d6f6e) and nixpkgs-unstable (467ce5a9f45aaf96110b41eb863a56866e1c2c3c).

I enabled vscode (vscodium) with extensions, but extensions are placed in wrong position `~/.vscode-oss/extensions/share/vscode/extensions/<extension_name>` instead of correct position `~/.vscode-oss/extensions/<extension_name>`. And only one of extensions from the list are actually installed.

Part of configuration:
```nix
{ pkgs, ... }:
{
  programs.vscode = {
    enable = true;
    package = pkgs.vscodium;
    extensions = [
      pkgs.vscode-extensions.bbenoist.Nix
      pkgs.vscode-extensions.vscodevim.vim # Only this is installed
    ];
  };
}
```

Note that this behavior is almost opposite from #1260 . 
I think we should swap behaviors of home-manager between 20.03 and unstable, which is, to strip the part `share/vscode/extensions` in nixpkgs-unstable but not do it in nixpkgs-20.03.

**Comment by worldofpeace** [comment]
ping @rycee

**Comment by rycee** [comment]
Sorry, I have no idea about this, I've never used vscode and don't know anything about it except that something changed in Nixpkgs unstable some time ago which broke the module and a fix was submitted in #1100. The fix accidentally got in the 20.03 branch, which I later reverted.

If this is some new breakage then I can only ask for a PR to fix it 🙁

**Comment by worldofpeace** [comment]
> Sorry, I have no idea about this, I've never used vscode and don't know anything about it except that something changed in Nixpkgs unstable some time ago which broke the module and a fix was submitted in #1100. The fix accidentally got in the 20.03 branch, which I later reverted.
> 
> If this is some new breakage then I can only ask for a PR to fix it slightly_frowning_face

Understood, I can look into it if I can manage it, since I use vscode.
This may relevant https://github.com/NixOS/nixpkgs/commit/0717848d7bb5adb01c7a9f293ed07676099de6f2
